flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -8,6 +8,15 @@ import '../../sentry_flutter.dart';
 /// See https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/
 const _navigationKey = 'navigation';
 
+typedef RouteNameExtractor = RouteSettings Function(RouteSettings settings);
+
+typedef AdditionalInfoProvider = Map<String, dynamic>? Function(
```

Comment:
I'm not quite sure about these names, any suggestions?

Reply:
I would change it from Extractor and Provider to Callback.
'Extractor' is very odd, and Provider is usually associated with other things in Flutter.

Reply:
Sounds good